src/main/kotlin/it/pagopa/wallet/scheduler/services/SchedulerLockService.kt

```diff
@@ -0,0 +1,90 @@
+package it.pagopa.wallet.scheduler.service
+
+import it.pagopa.wallet.scheduler.config.properties.RedisJobLockPolicyConfig
+import it.pagopa.wallet.scheduler.exceptions.LockNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.LockNotReleasedException
+import it.pagopa.wallet.scheduler.exceptions.SemNotAcquiredException
+import it.pagopa.wallet.scheduler.exceptions.SemNotReleasedException
+import java.util.concurrent.TimeUnit
+import org.redisson.api.RedissonReactiveClient
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+@Service
+class SchedulerLockService(
+    @Autowired private val redissonClient: RedissonReactiveClient,
+    @Autowired private val redisJobLockPolicyConfig: RedisJobLockPolicyConfig
+) {
+    private val logger: Logger = LoggerFactory.getLogger(javaClass)
+
+    fun acquireJobLock(jobName: String): Mono<Void> {
+        logger.info("Trying to acquire lock for job: {}", jobName)
+        return redissonClient
+            .getLock(redisJobLockPolicyConfig.getLockNameByJob(jobName))
+            .tryLock(
+                redisJobLockPolicyConfig.waitTimeSec,
+                redisJobLockPolicyConfig.ttlSec,
+                TimeUnit.SECONDS
+            )
+            .filter { it == true } // only lock acquired
+            .doOnSuccess { logger.info("Lock acquired for job: {}", jobName) }
+            .onErrorMap {
+                logger.error("Lock acquiring error for job: {}", jobName, it)
+                LockNotAcquiredException(jobName, it)
+            }
+            .switchIfEmpty(Mono.error(LockNotAcquiredException(jobName)))
+            .then()
+    }
```

Comment:
```suggestion
    fun acquireJobLock(jobName: String): Mono<Unit> {
        logger.info("Trying to acquire lock for job: {}", jobName)
        return redissonClient
            .getLock(redisJobLockPolicyConfig.getLockNameByJob(jobName))
            .tryLock(
                redisJobLockPolicyConfig.waitTimeSec,
                redisJobLockPolicyConfig.ttlSec,
                TimeUnit.SECONDS
            )
            .filter { it == true } // only lock acquired
            .doOnSuccess { logger.info("Lock acquired for job: {}", jobName) }
            .onErrorMap {
                logger.error("Lock acquiring error for job: {}", jobName, it)
                LockNotAcquiredException(jobName, it)
            }
            .switchIfEmpty(Mono.error(LockNotAcquiredException(jobName)))
            .thenReturn(Unit)
    }
```

You could use Unit here in order to resolve sonar cloud issue about using Void here with Kotlin.
This applies to other methods too.
What do you think?

Reply:
Done here 8732ddad4a60eb65146b3163be4a89bc36f89354